sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationSettingJsonConverter.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Azure.Data.AppConfiguration {
+    internal class ConfigurationSettingJsonConverter : JsonConverter<ConfigurationSetting>
+    {
+        public override ConfigurationSetting Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            return ConfigurationServiceSerializer.ReadSetting(ref reader);
+        }
+
+        public override void Write(Utf8JsonWriter writer, ConfigurationSetting value, JsonSerializerOptions options)
```

Comment:
Similarly for options here, should we throw if passed-in?

Reply:
We should throw `JsonException` for the case when serialization is not possible, but if `ConfigurationSetting` is available, we can serialize it. Do you have any other case in mind?

Reply:
I was just thinking we should use options, but see comment above -- I no longer think that.